src/java.base/share/classes/jdk/internal/io/JdkConsoleImpl.java

```diff
@@ -148,11 +148,13 @@ private void installShutdownHook() {
                             false /* only register if shutdown is not in progress */,
                             new Runnable() {
                                 public void run() {
-                                    try {
-                                        if (restoreEcho) {
-                                            echo(true);
-                                        }
-                                    } catch (IOException x) { }
+                                    synchronized(readLock) {
+                                        try {
+                                            if (restoreEcho) {
+                                                echo(true);
+                                            }
+                                        } catch (IOException _) { }
+                                    }
                                 }
                             });
         } catch (IllegalStateException e) {
```

Comment:
For consistency, this can also use `_`:
```suggestion
        } catch (IllegalStateException _) {
```